script.js

```diff
@@ -1,4 +1,341 @@
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // retutn the shuffled deck
+  shuffledDeck = cardDeck;
+  return shuffledDeck;
+};
+
+// deal 2 cards each to player and dealer from the shuffled deck
+var drawCard = function (cardDeck) {
+  var numOfCard = 2;
+  var cardCounter = 0;
+  while (cardCounter < numOfCard) {
+    playerHand.push(shuffledDeck.pop());
+    dealerHand.push(shuffledDeck.pop());
+    cardCounter += 1;
+  }
+
+  var dealedCards = `player Hand ${playerHand[0].name} of ${playerHand[0].suit} ${playerHand[1].name} of ${playerHand[1].suit} <br> dealer Hand  ${dealerHand[0].name} of ${dealerHand[0].suit}, ${dealerHand[1].name} of ${dealerHand[1].suit} `;
+  return dealedCards;
+};
+
+// compare player and dealer hand and return result.
+
+var checkResult = function () {
+  var winner = ``;
+  // draw condition
+  if (playersTotalPoints > 21 && dealersTotalPoints > 21) {
+    winner = `its a draw or busted: `;
+  }
+  // draw condition normal
+  else if (playersTotalPoints == dealersTotalPoints) {
+    winner = `its a draw `;
+  }
+  // player win by black jack
+  else if (
+    (playerHand[0].rank == 1 || playerHand[1].rank == 1) &&
+    (playerHand[0].rank >= 10 || playerHand[1].rank >= 10)
+  ) {
+    winner = `Black Jack ${player} wins`;
+  }
+  // dealer win by black jack
+  else if (
+    (dealerHand[0].rank == 1 || dealerHand[1].rank == 1) &&
+    (dealerHand[1].rank >= 10 || dealerHand[1].rank >= 10)
+  ) {
+    winner = `Black Jack ${dealer} wins`;
+  }
+  // player busted
+  else if (playersTotalPoints > 21 && dealersTotalPoints <= 21) {
+    winner = `congrats ${dealer} wins`;
+  }
+  // dealer busted
+  else if (playersTotalPoints <= 21 && dealersTotalPoints > 21) {
+    winner`congrats ${player} wins`;
+  }
+  // winning normal conditions
+  else if (playersTotalPoints > dealersTotalPoints) {
+    winner = ` congrats ${player} wins`;
+  } else if (playersTotalPoints < dealersTotalPoints) {
+    winner = `congrats ${dealer} wins`;
+  }
+  return winner;
+};
+
+// function to print player hand
+
+var printPlayerHand = function () {
+  var myOutputValue = ``;
+  var playerCardCounter = 0;
+  while (playerCardCounter < playerHand.length) {
+    myOutputValue =
+      myOutputValue +
+      ` ${playerHand[playerCardCounter].name} of ${playerHand[playerCardCounter].suit},`;
+    playerCardCounter += 1;
+  }
+  return myOutputValue;
+};
+
+// function to print dealer hand
+
+var printDealerHand = function () {
+  var myOutputValue = ``;
+  var dealerCardCounter = 0;
+  while (dealerCardCounter < dealerHand.length) {
+    myOutputValue =
+      myOutputValue +
+      `${dealerHand[dealerCardCounter].name} of ${dealerHand[dealerCardCounter].suit}`;
+    dealerCardCounter += 1;
+  }
+  return myOutputValue;
+};
+
+// keep track of player point
+
+var checkPlayerPoint = function () {
```

Comment:
Same here. You have the exact same function for both the player and computer. Try passing in arguments into the function